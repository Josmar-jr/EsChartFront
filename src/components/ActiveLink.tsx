import Link, { LinkProps } from 'next/link';
import { useRouter } from 'next/router';
import { cloneElement, ReactElement } from 'react';

type ActiveLink = LinkProps & {
  children: ReactElement;
  shouldMatchExactHref?: boolean;
};

const afterStyle =
  "after:content-[''] after:absolute after:h-10 after:w-[3.8px] after:rounded-r after:block after:top-[-4px] after:left-[-1px]";

export function ActiveLink({
  children,
  shouldMatchExactHref = false,
  ...rest
}: ActiveLink) {
  const { asPath } = useRouter();

  let isActive = false;

  if (shouldMatchExactHref && (asPath === rest.href || asPath === rest.as)) {
    isActive = true;
  }

  if (
    !shouldMatchExactHref &&
    (asPath.startsWith(String(rest.href)) || asPath.startsWith(String(rest.as)))
  ) {
    isActive = true;
  }

  return (
    <Link {...rest}>
      {cloneElement(children, {
        className: `hover:text-primary transition-all focus:ring focus:secondary outline-none inline-block px-[12px] w-full py-1 relative ${afterStyle} ${
          isActive ? 'text-primary sm:dark:after:bg-secondary dark:text-secondary sm:after:bg-primary' : 'text-slate-500 dark:text-slate-400'
        }`
      })}
    </Link>
  );
}
