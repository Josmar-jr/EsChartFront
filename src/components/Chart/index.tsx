import {
  CartesianGrid,
  Legend,
  Line,
  LineChart,
  Tooltip,
  XAxis,
  YAxis
} from 'recharts';

import { format } from 'date-fns';
import enUS from 'date-fns/locale/en-US';

import { CustomTooltip } from './CustomTooltip';

const data = [
  {
    date: format(new Date('02/24/2022'), 'dd/MM/yyyy', {
      locale: enUS
    }),
    pv: 2400
  },
  {
    date: new Date('01/24/2022').toString(),
    pv: 1398
  },
  {
    date: new Date('11/14/2022').toString(),
    pv: 9800
  },
  {
    date: new Date().toString(),
    pv: 3908
  },
  {
    date: new Date('06/16/2022').toString(),
    pv: 4800
  }
];

export function Chart() {
  return (
    <LineChart
      width={800}
      height={400}
      data={data}
      className="bg-blue-200 bg-opacity-20 rounded-md p-4"
      margin={{
        top: 5,
        right: 30,
        left: 20,
        bottom: 5
      }}
    >
      <CartesianGrid strokeDasharray="6 6" stroke="#d4d4d8" />
      <XAxis dataKey="date" format="Y-m-d" />
      <YAxis />
      <Tooltip
        offset={-90}
        content={<CustomTooltip />}
        coordinate={{
          x: 0,
          y: 0
        }}
      />
      <Legend />
      <Line
        type="monotone"
        dataKey="pv"
        stroke="#0095db"
        strokeWidth={3}
        activeDot={{ r: 6 }}
        dot={false}
      />
    </LineChart>
  );
}
