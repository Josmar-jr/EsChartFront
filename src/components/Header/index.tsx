import { List } from 'phosphor-react';
import { useDisclosure } from '../../contexts/DisclosureContext';
import { Menu } from './Menu';

export function Header() {
  const { onOpen } = useDisclosure();

  return (
    <header className="flex justify-between bg-neutral dark:bg-slate-900 border-b fixed top-0 left-0 border-slate-300 dark:border-slate-800 w-full py-2 px-4">
      <button
        onClick={onOpen}
        className="text-primary dark:text-secondary outline-none rounded-sm hover:brightness-95 focus:brightness-95 focus:ring"
      >
        <List size={26} />
      </button>
      <Menu />
    </header>
  );
}
