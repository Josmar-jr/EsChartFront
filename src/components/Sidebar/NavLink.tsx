import { LinkProps } from 'next/link';
import { ReactElement, ReactNode } from 'react';
import { ActiveLink } from '../ActiveLink';

type NavLinkProps = LinkProps & {
  children: ReactNode;
  href: string;
};

export function NavLink({ children, href, ...rest }: NavLinkProps) {
  return (
    <ActiveLink href={href} passHref>
      <a>{children}</a>
    </ActiveLink>
  );
}
