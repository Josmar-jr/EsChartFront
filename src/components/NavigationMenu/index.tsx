import * as NavMenuPrimitive from '@radix-ui/react-navigation-menu';
import { NavLink } from './NavLink';

export function NavigationMenu() {
  return (
    <NavMenuPrimitive.Root className="max-w-7xl mx-auto flex relative w-full pb-2">
      <NavMenuPrimitive.List className="flex justify-center items-center">
        <NavMenuPrimitive.Item>
          <NavMenuPrimitive.Trigger />
          <NavMenuPrimitive.Content>
            <NavMenuPrimitive.Link />
          </NavMenuPrimitive.Content>
        </NavMenuPrimitive.Item>

        <NavMenuPrimitive.Item>
          <NavLink href="/dashboard">Dashboard</NavLink>
        </NavMenuPrimitive.Item>
        <NavMenuPrimitive.Item>
          <NavLink href="/register">Register</NavLink>
        </NavMenuPrimitive.Item>
        <NavMenuPrimitive.Item>
          <NavLink href="/me/as">Settings</NavLink>
        </NavMenuPrimitive.Item>
      </NavMenuPrimitive.List>

      <NavMenuPrimitive.Viewport />
    </NavMenuPrimitive.Root>
  );
}
