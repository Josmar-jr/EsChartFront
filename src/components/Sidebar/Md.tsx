import {
  CaretDoubleRight,
  ChartPieSlice,
  Gear,
  SquaresFour
} from 'phosphor-react';
import { NavLink } from './NavLink';

interface MdProps {
  isOpen: boolean;
  onToggle: () => void;
}

export function Md({ isOpen, onToggle }: MdProps) {
  return (
    <aside
      className={`ease-linear z-50 duration-300 relative items-start flex flex-col h-screen ${
        isOpen ? 'w-12' : 'w-48'
      }`}
      role="menubar"
    >
      <div className="bg-neutral dark:bg-slate-900 w-full py-6 h-full border-slate-300 dark:border-slate-800 border-r">
        <div className="mx-auto flex items-center justify-center">
          <svg
            width="37"
            height="27"
            viewBox="0 0 37 27"
            fill="none"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14 26L18.8889 6H36L29.8889 26H14Z"
              fill="#034EA1"
              stroke="#034EA1"
            />
            <path
              d="M14 22.3086L13.3574 26H3.03516L3.67773 22.3086H14ZM8.17578 6.09375L4.7168 26H0.0410156L3.48633 6.09375H8.17578ZM14.1094 13.9961L13.4941 17.5645H4.49805L5.12695 13.9961H14.1094ZM16.8438 6.09375L16.2012 9.79883H5.83789L6.49414 6.09375H16.8438Z"
              fill="#0095DB"
            />
            <path
              d="M27.2891 19.2734L31.8691 19.2188C31.7415 20.7318 31.2539 22.0215 30.4062 23.0879C29.5677 24.1452 28.5059 24.9473 27.2207 25.4941C25.9447 26.041 24.582 26.3008 23.1328 26.2734C21.7565 26.2461 20.599 25.9635 19.6602 25.4258C18.7214 24.8789 17.9831 24.1497 17.4453 23.2383C16.9076 22.3177 16.5521 21.2878 16.3789 20.1484C16.2057 19 16.1966 17.8151 16.3516 16.5938L16.502 15.5137C16.6842 14.2467 17.0215 13.0299 17.5137 11.8633C18.0059 10.6875 18.6484 9.63932 19.4414 8.71875C20.2435 7.79818 21.196 7.07812 22.2988 6.55859C23.4017 6.02995 24.6504 5.7793 26.0449 5.80664C27.5306 5.83398 28.793 6.14388 29.832 6.73633C30.8802 7.32878 31.6914 8.1582 32.2656 9.22461C32.849 10.291 33.1771 11.5488 33.25 12.998L28.5879 12.9844C28.6152 12.2826 28.5469 11.681 28.3828 11.1797C28.2188 10.6784 27.9271 10.291 27.5078 10.0176C27.0977 9.73503 26.5189 9.58008 25.7715 9.55273C24.9603 9.52539 24.2812 9.69401 23.7344 10.0586C23.1966 10.4141 22.7591 10.8926 22.4219 11.4941C22.0846 12.0957 21.8249 12.7474 21.6426 13.4492C21.4694 14.151 21.3372 14.8301 21.2461 15.4863L21.1094 16.6074C21.0456 17.1452 20.9863 17.7513 20.9316 18.4258C20.877 19.0911 20.8997 19.7337 21 20.3535C21.1003 20.9642 21.3327 21.4746 21.6973 21.8848C22.0618 22.2949 22.6315 22.5137 23.4062 22.541C24.1354 22.5592 24.7689 22.4453 25.3066 22.1992C25.8444 21.944 26.2773 21.5703 26.6055 21.0781C26.9427 20.5768 27.1706 19.9753 27.2891 19.2734Z"
              fill="white"
            />
          </svg>
        </div>

        <button
          onClick={onToggle}
          className={`w-8 h-8 hover:brightness-95 ease-linear dark:border dark:bg-slate-900 dark:border-slate-700 outline-none focus:brightness-95 focus:ring focus:secondary duration-300 absolute bg-neutral top-16 left-[174px] ${
            isOpen && 'translate-x-[-144px]'
          } ${
            !isOpen && 'rotate-180'
          } drop-shadow-md rounded-full flex items-center justify-center`}
          role="button"
        >
          <CaretDoubleRight size={16} weight="bold" />
        </button>

        <nav className="mt-28">
          <strong className="text-xs ml-4 translate-y-[-10px] inline-block">
            {!isOpen && 'Menu'}
          </strong>
          <ul className="flex gap-5 flex-col text-slate-500 font-medium">
            <li>
              <NavLink href="/dashboard">
                <span className="flex items-center gap-2">
                  <SquaresFour size={24} />
                  {!isOpen && 'Dashboard'}
                </span>
              </NavLink>
            </li>
            <li>
              <NavLink href="/2">
                <span className="flex items-center gap-2">
                  <ChartPieSlice size={24} />
                  {!isOpen && 'Chart'}
                </span>
              </NavLink>
            </li>
            <li>
              <NavLink href="/algo">
                <span className="flex items-center gap-2">
                  <Gear size={24} />
                  {!isOpen && 'Settings'}
                </span>
              </NavLink>
            </li>
          </ul>
        </nav>
      </div>
    </aside>
  );
}
