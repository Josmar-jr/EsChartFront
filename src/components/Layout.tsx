import { ReactNode } from 'react';
import { DisclosureProvider } from '../contexts/DisclosureContext';
import { Header } from './Header';
import { Sidebar } from './Sidebar';

interface LayoutProps {
  children: ReactNode;
}

export function Layout({ children }: LayoutProps) {
  return (
    <>
      <DisclosureProvider>
        <>
          <Sidebar />
          <Header />
        </>
      </DisclosureProvider>
      <main className="mt-16 ml-4 md:ml-6">{children}</main>
    </>
  );
}
