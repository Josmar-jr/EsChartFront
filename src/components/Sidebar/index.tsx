import { useScreenSize } from '../../hooks/useWindowSize';
import { useDisclosure } from '../../contexts/DisclosureContext';

import { Sm } from './Sm';
import { Md } from './Md';

export function Sidebar() {
  const { isOpen, onToggle, onClose } = useDisclosure();

  const { screenSize } = useScreenSize();

  return screenSize === 'sm' ? (
    <Sm isOpen={isOpen} onClose={onClose} />
  ) : (
    <Md isOpen={isOpen} onToggle={onToggle} />
  );
}
