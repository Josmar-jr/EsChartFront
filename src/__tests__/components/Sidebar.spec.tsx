import { render, screen, cleanup, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Sidebar } from '../../components/Sidebar';

afterEach(() => {
  cleanup();
});

jest.mock('next/router', () => {
  return {
    useRouter() {
      return {
        asPath: '/test'
      };
    }
  };
});

describe('Input Component', () => {
  it('should render correctly', () => {
    render(<Sidebar />);

    const sidebar = screen.getByRole('menubar');

    expect(sidebar).toBeInTheDocument();
  });

  it('should start expand the sidebar', () => {
    render(<Sidebar />);

    const sidebar = screen.getByRole('menubar');
    const button = screen.getByRole('button');

    expect(sidebar).toHaveClass('w-48');
    expect(sidebar).not.toHaveClass('w-12');
    expect(button).toHaveClass('rotate-180');
  });

  it('should compress the sidebar when clicking the button ', () => {
    render(<Sidebar />);

    const sidebar = screen.getByRole('menubar');
    const button = screen.getByRole('button');

    fireEvent.click(button);

    expect(sidebar).not.toHaveClass('w-48');
    expect(sidebar).toHaveClass('w-12');
    expect(button).toHaveClass('translate-x-[-124px]');
  });

  it('should match snapshot', () => {
    const tree = renderer.create(<Sidebar />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
