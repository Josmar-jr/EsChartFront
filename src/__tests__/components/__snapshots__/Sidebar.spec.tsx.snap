// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component should match snapshot 1`] = `
<aside
  className="border ease-linear duration-300 border-l relative py-6 items-start border-slate-300 flex flex-col h-screen w-48"
  role="menubar"
>
  <div
    className="mx-auto flex items-center justify-center"
  >
    <svg
      fill="none"
      height="27"
      viewBox="0 0 37 27"
      width="37"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M14 26L18.8889 6H36L29.8889 26H14Z"
        fill="#034EA1"
        stroke="#034EA1"
      />
      <path
        d="M14 22.3086L13.3574 26H3.03516L3.67773 22.3086H14ZM8.17578 6.09375L4.7168 26H0.0410156L3.48633 6.09375H8.17578ZM14.1094 13.9961L13.4941 17.5645H4.49805L5.12695 13.9961H14.1094ZM16.8438 6.09375L16.2012 9.79883H5.83789L6.49414 6.09375H16.8438Z"
        fill="#0095DB"
      />
      <path
        d="M27.2891 19.2734L31.8691 19.2188C31.7415 20.7318 31.2539 22.0215 30.4062 23.0879C29.5677 24.1452 28.5059 24.9473 27.2207 25.4941C25.9447 26.041 24.582 26.3008 23.1328 26.2734C21.7565 26.2461 20.599 25.9635 19.6602 25.4258C18.7214 24.8789 17.9831 24.1497 17.4453 23.2383C16.9076 22.3177 16.5521 21.2878 16.3789 20.1484C16.2057 19 16.1966 17.8151 16.3516 16.5938L16.502 15.5137C16.6842 14.2467 17.0215 13.0299 17.5137 11.8633C18.0059 10.6875 18.6484 9.63932 19.4414 8.71875C20.2435 7.79818 21.196 7.07812 22.2988 6.55859C23.4017 6.02995 24.6504 5.7793 26.0449 5.80664C27.5306 5.83398 28.793 6.14388 29.832 6.73633C30.8802 7.32878 31.6914 8.1582 32.2656 9.22461C32.849 10.291 33.1771 11.5488 33.25 12.998L28.5879 12.9844C28.6152 12.2826 28.5469 11.681 28.3828 11.1797C28.2188 10.6784 27.9271 10.291 27.5078 10.0176C27.0977 9.73503 26.5189 9.58008 25.7715 9.55273C24.9603 9.52539 24.2812 9.69401 23.7344 10.0586C23.1966 10.4141 22.7591 10.8926 22.4219 11.4941C22.0846 12.0957 21.8249 12.7474 21.6426 13.4492C21.4694 14.151 21.3372 14.8301 21.2461 15.4863L21.1094 16.6074C21.0456 17.1452 20.9863 17.7513 20.9316 18.4258C20.877 19.0911 20.8997 19.7337 21 20.3535C21.1003 20.9642 21.3327 21.4746 21.6973 21.8848C22.0618 22.2949 22.6315 22.5137 23.4062 22.541C24.1354 22.5592 24.7689 22.4453 25.3066 22.1992C25.8444 21.944 26.2773 21.5703 26.6055 21.0781C26.9427 20.5768 27.1706 19.9753 27.2891 19.2734Z"
        fill="white"
      />
    </svg>
  </div>
  <button
    className="w-8 h-8 hover:brightness-95 ease-linear outline-none focus:brightness-95 focus:ring focus:secondary duration-300 absolute bg-neutral top-16 left-[156px] false rotate-180 drop-shadow-md rounded-full flex items-center justify-center"
    onClick={[Function]}
    role="button"
  >
    <svg
      fill="currentColor"
      height={16}
      viewBox="0 0 256 256"
      width={16}
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        fill="none"
        height="256"
        width="256"
      />
      <polyline
        fill="none"
        points="56 48 136 128 56 208"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="24"
      />
      <polyline
        fill="none"
        points="136 48 216 128 136 208"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="24"
      />
    </svg>
  </button>
  <nav
    className="mt-28"
  >
    <strong
      className="text-xs ml-4 translate-y-[-10px] inline-block"
    >
      Menu
    </strong>
    <ul
      className="flex gap-5 flex-col text-slate-500 font-medium"
    >
      <li
        className=""
      >
        <a
          className="hover:text-primary transition-all focus:ring focus:secondary outline-none inline-block px-[12px] w-full py-1 relative after:content-[''] after:absolute after:h-10 after:w-[3.8px] after:rounded-r after:block after:top-[-4px] after:left-[-1px] text-slate-500"
          href="/dashboard"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <span
            className="flex items-center gap-2"
          >
            <svg
              fill="currentColor"
              height={24}
              viewBox="0 0 256 256"
              width={24}
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="256"
                width="256"
              />
              <rect
                fill="none"
                height="64"
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="16"
                width="64"
                x="48"
                y="48"
              />
              <rect
                fill="none"
                height="64"
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="16"
                width="64"
                x="144"
                y="48"
              />
              <rect
                fill="none"
                height="64"
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="16"
                width="64"
                x="48"
                y="144"
              />
              <rect
                fill="none"
                height="64"
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="16"
                width="64"
                x="144"
                y="144"
              />
            </svg>
            Dashboard
          </span>
        </a>
      </li>
      <li>
        <a
          className="hover:text-primary transition-all focus:ring focus:secondary outline-none inline-block px-[12px] w-full py-1 relative after:content-[''] after:absolute after:h-10 after:w-[3.8px] after:rounded-r after:block after:top-[-4px] after:left-[-1px] text-slate-500"
          href="/2"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <span
            className="flex items-center gap-2"
          >
            <svg
              fill="currentColor"
              height={24}
              viewBox="0 0 256 256"
              width={24}
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="256"
                width="256"
              />
              <line
                fill="none"
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="16"
                x1="128"
                x2="128"
                y1="128"
                y2="32"
              />
              <line
                fill="none"
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="16"
                x1="211.1"
                x2="44.9"
                y1="80"
                y2="176"
              />
              <path
                d="M33.6,145.6A92.9,92.9,0,0,1,32,128,95.9,95.9,0,0,1,96,37.5v72Z"
                fill="none"
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="16"
              />
              <path
                d="M128,32A96,96,0,1,1,45.2,176.6"
                fill="none"
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="16"
              />
            </svg>
            Chart
          </span>
        </a>
      </li>
      <li>
        <a
          className="hover:text-primary transition-all focus:ring focus:secondary outline-none inline-block px-[12px] w-full py-1 relative after:content-[''] after:absolute after:h-10 after:w-[3.8px] after:rounded-r after:block after:top-[-4px] after:left-[-1px] text-slate-500"
          href="/algo"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <span
            className="flex items-center gap-2"
          >
            <svg
              fill="currentColor"
              height={24}
              viewBox="0 0 256 256"
              width={24}
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="256"
                width="256"
              />
              <circle
                cx="128"
                cy="128"
                fill="none"
                r="48"
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="16"
              />
              <path
                d="M183.7,65.1q3.8,3.5,7.2,7.2l27.3,3.9a103.2,103.2,0,0,1,10.2,24.6l-16.6,22.1s.3,6.8,0,10.2l16.6,22.1a102.2,102.2,0,0,1-10.2,24.6l-27.3,3.9s-4.7,4.9-7.2,7.2l-3.9,27.3a103.2,103.2,0,0,1-24.6,10.2l-22.1-16.6a57.9,57.9,0,0,1-10.2,0l-22.1,16.6a102.2,102.2,0,0,1-24.6-10.2l-3.9-27.3q-3.7-3.5-7.2-7.2l-27.3-3.9a103.2,103.2,0,0,1-10.2-24.6l16.6-22.1s-.3-6.8,0-10.2L27.6,100.8A102.2,102.2,0,0,1,37.8,76.2l27.3-3.9q3.5-3.7,7.2-7.2l3.9-27.3a103.2,103.2,0,0,1,24.6-10.2l22.1,16.6a57.9,57.9,0,0,1,10.2,0l22.1-16.6a102.2,102.2,0,0,1,24.6,10.2Z"
                fill="none"
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="16"
              />
            </svg>
            Settings
          </span>
        </a>
      </li>
    </ul>
  </nav>
</aside>
`;
